contracts/src/libraries/LPMath.sol

```diff
@@ -127,6 +127,124 @@ library LPMath {
+    /// @param _vaultSharePrice The vault share price.
+    /// @param _timeStretch The time stretch.
+    /// @param _checkpointExposure The checkpoint exposure.
+    /// @param _longExposure The global long exposure.
+    /// @return A flag indicating if the price discovery is valid.
+    function verifyPriceDiscovery(
+        uint256 _shareReserves,
+        int256 _shareAdjustment,
+        uint256 _bondReserves,
+        uint256 _minimumShareReserves,
+        uint256 _initialVaultSharePrice,
+        uint256 _vaultSharePrice,
+        uint256 _timeStretch,
+        int256 _checkpointExposure,
+        uint256 _longExposure
+    ) external pure returns (bool) {
+        // Calculate the share payment and bond proceeds of opening the
+        // largest possible long on the YieldSpace curve. This does not
+        // include fees.
+        (uint256 effectiveShareReserves, bool success) = HyperdriveMath
+            .calculateEffectiveShareReservesSafe(
+                _shareReserves,
+                _shareAdjustment
+            );
+        if (!success) {
+            return false;
+        }
+        uint256 maxSharePayment;
+        (maxSharePayment, success) = YieldSpaceMath.calculateMaxBuySharesInSafe(
+            effectiveShareReserves,
+            _bondReserves,
+            ONE - _timeStretch,
+            _vaultSharePrice,
+            _initialVaultSharePrice
+        );
+
+        uint256 maxBondProceeds;
+        (maxBondProceeds, success) = YieldSpaceMath.calculateMaxBuyBondsOutSafe(
+            effectiveShareReserves,
+            _bondReserves,
+            ONE - _timeStretch,
+            _vaultSharePrice,
+            _initialVaultSharePrice
+        );
+
+        // If one fails they should both be zero.
+        if (maxSharePayment == 0 || maxBondProceeds == 0) {
+            maxSharePayment = 0;
+            maxBondProceeds = 0;
+        }
+
+        // Calculate the pool's solvency after opening the max long. This
+        // doesn't account for fees, which is fine since this will be more
+        // conservative.
+        uint256 shareReserves = _shareReserves + maxSharePayment;
+        uint256 longExposure = calculateLongExposure(
+            _longExposure,
+            _checkpointExposure,
+            _checkpointExposure + maxBondProceeds.toInt256()
+        );
+
+        // If the pool isn't solvent after opening the max long, then we
+        // prevent the liquidity from being added since it will cause issues
+        // with price discovery. We know that when: cz <= e_l + c * z_min
+        // is within some epilson of being true, the pool is still solvent,
+        // will have trouble price discovering back to 0%. We add an additional
+        // c * z_min to the right hand side to account for this epsilon.
+        if (
+            shareReserves.mulDown(_vaultSharePrice) <=
+            longExposure + 2 * _minimumShareReserves.mulUp(_vaultSharePrice)
+        ) {
+            return false;
+        }
```

Comment:
I think beside this check we should also check that the updated point has a spot price close to one.

Reply:
We acknowledge this, but we've decided not to include this check because (1) fees will result in the spot price not being equal to 1 and (2) the method that we are using is mathematically equivalent when there are no fees.